packages/graph/hash_graph/lib/graph/src/api/rest/data_type.rs

```diff
@@ -118,6 +120,24 @@ async fn create_data_type<P: StorePool + Send>(
         .map(Json)
 }
 
+#[utoipa::path(
+    post,
+    path = "/data-types/query",
+    request_body = Expression,
+    tag = "DataType",
+    responses(
+        (status = 200, content_type = "application/json", description = "List of all data types matching the provided query", body = [PersistedDataType]),
+
+        (status = 500, description = "Store error occurred"),
```

Comment:
Do we really not have any other error types? What if the query is malformed or invalid?

Reply:
Currently, it only supports a return code of `500`. `404` is not supported as an empty list would then be returned (in comparison to `get_data_type`

Reply:
Would it be easier to give some better codes here? I appreciate we're revisiting error handling ASAP but if we have some available it'd be good to expose them.

The queries have some resolution failures right? If expressions aren't booleans or something. I'm assuming we'd want to treat those as 422's?

Reply:
I was able to add `422` codes: 29c271ab582f0854fa8c4d5a1551a69ab906a51a